core/src/test/java/org/apache/calcite/test/RelOptRulesTest.java

```diff
@@ -2952,7 +2952,7 @@ private void checkReduceNullableToNotNull(ReduceExpressionsRule rule) {
     // in addition to the casts.
     checkPlanning(program,
         "select * from emp "
-            + "where cast((empno + (10/2)) as int) = 13");
+            + "where cast((empno + (10/2)) as double) = 13");
```

Comment:
why ? isn't 13 int?

Reply:
13 is int, but with implicit coercion, the `/` operator always returns double now.

Reply:
I don't think we should cast integer (IIRC) into decimal for DIV, unless there is at least one decimal operand.